src/precice/impl/DataContext.hpp

```diff
@@ -91,25 +91,25 @@ class DataContext {
   DataContext(mesh::PtrData data, mesh::PtrMesh mesh);
 
   /// Defines the mapping associated to this DataContext. A DataContext may also exist without a mapping.
-  MappingContext _mappingContext;
+  std::vector<MappingContext> _mappingContext;
 
-  /// Data this participant will write to and read from
+  /// Unique data this context is associated with
   mesh::PtrData _providedData;
 
   /// If a mapping exists, mesh::PtrData the mapping maps from.
-  mesh::PtrData _fromData;
+  std::vector<mesh::PtrData> _fromData;
 
   /// If a mapping exists, mesh::PtrData the mapping maps from.
-  mesh::PtrData _toData;
+  std::vector<mesh::PtrData> _toData;
 
   /**
-   * @brief Helper to set _mappingContext, _fromData and _toData.
+   * @brief Helper to add _mappingContext, _fromData and _toData.
```

Comment:
This comment is misleading. You don't add a `_mappingContext`, but you add a `mappingContext` to the `_mappingContexts` container, if I understand this correctly.